### PR TITLE
feat!: hex encode `id.String()` output

### PR DESCRIPTION
**Pull request by rootulp**
This PR contains a breaking change that implements https://github.com/celestiaorg/nmt/pull/201#issuecomment-1564414023. Relevant https://github.com/celestiaorg/nmt/issues/200

### CODE REVIEW

**Comment by staheri14** [comment]
@rootulp Is the PR ready for review?  It says it is in draft mode :)

**Comment by rootulp** [comment]
It is not bc this PR is failing a test.

**Comment by rootulp** [comment]
Fixed failing test via https://github.com/celestiaorg/nmt/pull/203

**Review by walldiss — APPROVED** [review]

File: namespace/id.go [hunk 1]
```diff
@@ -27,12 +27,7 @@ func (nid ID) Size() IDSize {
 	return IDSize(len(nid))
 }
 
-// String stringifies the nid.
+// String returns the hexadecimal encoding of the nid.
```
Comment (staheri14) [inline namespace/id.go:30]: [optional] Thank you for implementing the additional test for `string(nid)`. To ensure full understanding, I recommend we include a remark at this point to clarify that `string(nid)` is not equivalent to `nid.String()`. This clarification is important, as these two can easily be mistaken for each other.

**Review by staheri14 — APPROVED** [review]

File: namespace/id.go (hunk 1 above)
Comment (rootulp) [reply namespace/id.go]: Resolved by https://github.com/celestiaorg/nmt/pull/202/commits/d71f15fbdb7189d8fd39d54bfc28d7f11296f1bb

**Review by staheri14 — DISMISSED** [review]

**Review by staheri14 — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**